include/envoy/upstream/health_checker.h

```diff
@@ -21,7 +25,8 @@ class HealthChecker {
    * @param changed_state supplies whether the health check resulted in a host moving from healthy
    *                       to not healthy or vice versa.
    */
-  typedef std::function<void(HostSharedPtr host, bool changed_state)> HostStatusCb;
+  typedef std::function<void(HostSharedPtr host, HealthState state, HealthTransition changed_state)>
```

Comment:
nit: please update doc comments.

Q: What the reasoning behind passing state here? State can be obtained from the host directly. (Mostly just wondering if it's worth the param churn here when the data can be gotten easily from the host. I also see you not using it in all cases.)

Reply:
now that you mentioned it, you're right, there's no point doing that. I'll get rid of the additional parameter. I'll leave `changed_state` as `HealthTransition`, though, since it not only documents the API better but the type-safety also makes it harder to misuse the API.